wagtail/wagtailcore/models.py

```diff
@@ -193,7 +193,7 @@ def __init__(cls, name, bases, dct):
 
         if 'template' not in dct:
             # Define a default template path derived from the app name and model name
-            cls.template = "%s/%s.html" % (cls._meta.app_label, camelcase_to_underscore(name))
+            cls.template = "%s/%s.%s" % (cls._meta.app_label, camelcase_to_underscore(name), 'html' if not hasattr(settings, 'MODEL_TEMPLATE_TYPE') else settings.MODEL_TEMPLATE_TYPE)
```

Comment:
As a setting `'html' if not hasattr(settings, 'MODEL_TEMPLATE_TYPE') else settings.MODEL_TEMPLATE_TYPE` would be neater as `getattr(settings, 'MODEL_TEMPLATE_TYPE', 'html')`.

However an undocumented, non-wagtail-scoped settings attribute probably isn't the best way to override the template extension. The template can be overridden by setting the template on the concrete Page class, but you do then need to specify the full template path (dir and file basename as well as the extension) which isn't terribly DRY when you just want to change the extension.

Maybe a documented optional `template_extension =  'html'` attribute for concrete `Page` classes is better with `PageBase` having something like:
```
if 'template' not in dct:
    template_extension = dct['template_extension'] if 'template_extension' in dct else 'html'
    cls.template = "%s/%s.%s" % (cls._meta.app_label, camelcase_to_underscore(name), template_extension)
```

Still needs the template_extension to be set on each concrete `Page` model with a non-html template extension, but it does make it explicit and maybe preferable to another `WAGTAIL*` setting. What do you think @gasman?